src/pudl/transform/ferc1.py

```diff
@@ -1697,6 +1721,33 @@ def apply_xbrl_calc_fixes(self: Self, tbl_meta: pd.DataFrame) -> pd.DataFrame:
                         },
                     },
                 ],
+                "sales_to_ultimate_consumers": [
+                    # these next two fixes are related. we're replacing
+                    # commercial_and_industrial_sales which is from a different table
+                    # with two swappable components from this table. There is no way rn
+                    # to say swap thing A for thing B & C for two so we swap A for B and
+                    # then add C.
```

Comment:
I don't understand what this comment is saying. What are "two swappable components?"

We're deleting one component and adding two components, right?

It seems like it might be simpler to have the more atomic "addition" and "deletion" operations rather than always defaulting to "replace" which is a composite of a deletion and an addition.  We could define a replacement method in terms of deletion and addition methods.

What happens when you replace a component that has a list of source tables with a component that doesn't have a list of source tables?

Reply:
I think what's happening is

```py
# Replace commercial_and_industrial_sales which is reported in
# electricity_sales_by_rate_schedule_ferc1 with two components
# (small_or_commercial_sales... and large_or_industrial_sales...)
# from this table which should sum to the same amount as the value
# being replaced.
```

Reply:
code wise i hard agree that having "additions" and "deletions" would make way more sense. As a human conceptually it makes way more sense to say "replace A for B" (and A or B can be a null element which results in "add B" or "delete A" respective).

i only added that comment bc rn the fixes are setup as "replace A for B" not "replace A for B & C" which is what needed to happen here. but that needed to be added as "replace A for B" & "add C".

i'd love a deeper structural change for these calcs but that is for  #2605.